airbyte-integrations/connectors/source-apple-search-ads/source_apple_search_ads/apple_search_ads.yaml

```diff
@@ -0,0 +1,228 @@
+
+  stream_slicer_adgroup:
+    type: SubstreamSlicer
+    parent_stream_configs:
+      - stream: "*ref(definitions.adgroups_stream)"
+        parent_key: "id"
+        stream_slice_field: "adgroup_id"
+
+  keywords_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "keywords"
+      primary_key: "id"
+    retriever:
+      $ref: "*ref(definitions.base_stream.retriever)"
+      requester:
+        $ref: "*ref(definitions.requester)"
+        path: "/campaigns/{{ stream_slice.parent_slice.campaign_id }}/adgroups/{{ stream_slice.adgroup_id }}/targetingkeywords"
+      stream_slicer:
+        type: CartesianProductStreamSlicer
+        stream_slicers:
+          - "*ref(definitions.stream_slicer_adgroup)"
+
+  stream_slicer_date:
+    type: "DatetimeStreamSlicer"
+    start_datetime:
+      datetime: "{{ config['start_date'] }}"
+      datetime_format: "%Y-%m-%d"
+    end_datetime:
+      datetime: "{{ config['end_date'] or today_utc() }}"
+      datetime_format: "%Y-%m-%d"
+    step: "1d"
+    lookback_window: "30d"
+    datetime_format: "%Y-%m-%d"
+    cursor_field: "{{ options['stream_cursor_field'] }}"
+
+  report_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+      record_selector:
+        extractor:
+          type: DpathExtractor
+          field_pointer: ["data", "reportingDataResponse", "row"]
+      paginator:
+        type: NoPagination
+      stream_slicer:
+        type: CartesianProductStreamSlicer
+        stream_slicers:
+          - "*ref(definitions.stream_slicer_date)"
+      requester:
+        $ref: "*ref(definitions.requester)"
+        http_method: "POST"
+        request_options_provider:
+          request_headers:
+            X-AP-Context: "orgId={{ config['org_id'] }}"
+          request_body_json:
+            startTime: "{{ stream_slice['start_time'] }}"
+            endTime: "{{ stream_slice['end_time'] }}"
+            timeZone: "UTC"
+            granularity: "{{ options.granularity }}"
+            selector: '{
+              "orderBy": [
+                {
+                  "field": "countryOrRegion",
+                  "sortOrder": "ASCENDING"
+                }
+              ]
+            }'
+            groupBy: "[ 'countryOrRegion' ]"
+
+  campaigns_report_daily_stream:
+    $ref: "*ref(definitions.report_stream)"
+    $options:
+      name: "campaigns_report_daily"
+      stream_cursor_field: "date"
+      granularity: "DAILY"
+      primary_key: [["date"], ["campaignId"]]
+    retriever:
+      $ref: "*ref(definitions.report_stream.retriever)"
+      requester:
+        $ref: "*ref(definitions.report_stream.retriever.requester)"
+        path: "/reports/campaigns"
+    transformations:
+      - type: AddFields
+        fields:
+          - path: [ "campaignId" ]
+            value: "{{ record['metadata']['campaignId'] }}"
+          - path: [ "date" ]
+            value: "{{ stream_slice['start_time'] }}"
```

Comment:
As nested cursors are not supported yet for `stream_cursor_field` (and even for `primary_key` for normalization?) I am adding them as new fields at the root of the each stream (thanks @brianjlai for the tip 👌)